### ISSUE TITLE
res.locals is not available if layout is defined

### ISSUE THREAD

**Issue by yaneony** [fastify/point-of-view#207]
As in title said, res.locals definition is not available from views if layout is defined. 

```js
const fastify = require('fastify')();

(async () => {
  await fastify.register(require('middie'));
  fastify.register(require('point-of-view'), { engine: { ejs: require('ejs') }, layout: 'layout', viewExt: 'ejs' });
})();

(async () => {
  await fastify.listen(4040, '127.0.0.1');
})();

fastify.addHook('preHandler', async function (req, res) {
  res.locals = {};
  res.locals.user = 'anonymous';
});

fastify.get('/', async (req, res) => {
  return res.view('template');
});
```

So, user variable will be available in layout.ejs, but it's not available in template.ejs file.

Is there any way to make locals available to nested views also too?!

**Comment by yaneony** [comment]
Okay, checking out the code, found that problem is with `withLayout` function. First call is not passing any local data thru.

Old code:
```JS
  function withLayout (render) {
    if (layoutFileName) {
      return function (page, data, opts) {
        const that = this

        render.call({
          getHeader: () => { },
          header: () => { },
          send: (result) => {
            if (result instanceof Error) {
              throw result
            }

            data = Object.assign((data || {}), { body: result })

            render.call(that, layoutFileName, data, opts)
          }
        }, page, data, opts)
      }
    }

    return render
  }```
  
  Added new single line before `render.call` did the magic. Not sure about performance and other stuff... Can anyone check if that good solution?

New code:
```JS
  function withLayout (render) {
    if (layoutFileName) {
      return function (page, data, opts) {
        const that = this

        data = Object.assign({}, defaultCtx, this.locals, data) // THIS LINE ADDED
        
        render.call({
          getHeader: () => { },
          header: () => { },
          send: (result) => {
            if (result instanceof Error) {
              throw result
            }

            data = Object.assign((data || {}), { body: result })

            render.call(that, layoutFileName, data, opts)
          }
        }, page, data, opts)
      }
    }

    return render
  }
```

Thanks!

**Comment by mcollina** [comment]
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by yaneony** [comment]
The question is if that's good solution...

**Comment by mcollina** [comment]
yes it is!

**Comment by yaneony** [comment]
Proper code is

```JS
  function withLayout (render) {
    if (layoutFileName) {
      return function (page, data, opts) {
        const that = this

        data = Object.assign({}, defaultCtx, this.locals, data)
        
        render.call({
          getHeader: () => { },
          header: () => { },
          send: (result) => {
            if (result instanceof Error) {
              throw result
            }

            data = Object.assign(data, { body: result })

            render.call(that, layoutFileName, data, opts)
          }
        }, page, data, opts)
      }
    }

    return render
  }
```

If someone could commit that, that would be nice. I'm really short on time...

**Comment by yaneony** [comment]
The proper code is:

```JS
  function withLayout (render) {
    if (layoutFileName) {
      return function (page, data, opts) {
        const that = this

        data = Object.assign({}, defaultCtx, this.locals, data)
        
        render.call({
          getHeader: () => { },
          header: () => { },
          send: (result) => {
            if (result instanceof Error) {
              throw result
            }

            data = Object.assign(data || {}), { body: result })

            render.call(that, layoutFileName, data, opts)
          }
        }, page, data, opts)
      }
    }

    return render
  }
```

If someone could commit that one, that would be nice. I'm short on time, lost a lot to understand why that was not working... And deadline is near... Thanks!

**Comment by yaneony** [comment]
Fixed with #209.
